homeassistant/components/google_domains/__init__.py

```diff
@@ -42,35 +42,47 @@ async def async_setup(hass, config):
 
     session = hass.helpers.aiohttp_client.async_get_clientsession()
 
-    result = await _update_google_domains(
-        hass, session, domain, user, password, timeout
+    address = await _update_google_domains(
+        hass, session, domain, user, password, timeout, None
     )
 
-    if not result:
+    if address is None:
         return False
 
     async def update_domain_interval(now):
-        """Update the Google Domains entry."""
-        await _update_google_domains(hass, session, domain, user, password, timeout)
+        nonlocal address
+        address = await _update_google_domains(
+            hass, session, domain, user, password, timeout, address
+        )
 
     hass.helpers.event.async_track_time_interval(update_domain_interval, INTERVAL)
 
     return True
 
 
-async def _update_google_domains(hass, session, domain, user, password, timeout):
-    """Update Google Domains."""
-    url = f"https://{user}:{password}@domains.google.com/nic/update"
+async def _update_google_domains(
+    hass, session, domain, user, password, timeout, address
+):
+    try:
+        # Check IP address
+        with async_timeout.timeout(timeout):
+            resp = await session.get("https://api.ipify.org")
+            body = await resp.text()
+
+            if body == address:
```

Comment:
This may be leaning into style/readability preference: i think having a separate function for "get my ip" than "update my ip" would make sense then moving this logic into the callers.  That is, `update_domain_interval` checks if the address has changed before calling `_update_google_domains``, would be easier to reason about than a `None` argument to change behavior.

Reply:
I'm inclined to agree. I was trying to make as few changes as possible, but for the sake of readability I'll refactor.